File: CapstoneSystemV4/pwa-sample/generate-icons.js
// Simple script to generate PWA icons
// Run with: node generate-icons.js
// Requires: npm install canvas (or use the HTML generator instead)

const fs = require('fs');
const path = require('path');

// Create simple SVG icons and convert them
// Since canvas might not be available, we'll create SVG files instead
// which can be converted to PNG using online tools or ImageMagick

function createSVGIcon(size) {
    return `<?xml version="1.0" encoding="UTF-8"?>
<svg width="${size}" height="${size}" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <linearGradient id="grad" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#667eea;stop-opacity:1" />
      <stop offset="100%" style="stop-color:#764ba2;stop-opacity:1" />
    </linearGradient>
  </defs>
  <rect width="${size}" height="${size}" fill="url(#grad)" rx="${size * 0.1}"/>
  <text x="50%" y="50%" font-size="${size * 0.4}" text-anchor="middle" dominant-baseline="central" fill="white">🚀</text>
</svg>`;
}

// Try to use canvas if available, otherwise create SVG
try {
    const { createCanvas } = require('canvas');
    
    function generatePNGIcon(size) {
        const canvas = createCanvas(size, size);
        const ctx = canvas.getContext('2d');
        
        // Gradient background
        const gradient = ctx.createLinearGradient(0, 0, size, size);
        gradient.addColorStop(0, '#667eea');
        gradient.addColorStop(1, '#764ba2');
        ctx.fillStyle = gradient;
        ctx.fillRect(0, 0, size, size);
        
        // Draw rocket emoji (as text)
        ctx.fillStyle = 'white';
        ctx.font = `bold ${size * 0.4}px Arial`;
        ctx.textAlign = 'center';
        ctx.textBaseline = 'middle';
        ctx.fillText('🚀', size / 2, size / 2);
        
        return canvas.toBuffer('image/png');
    }
    
    // Generate PNG icons
    fs.writeFileSync('icon-192.png', generatePNGIcon(192));
    fs.writeFileSync('icon-512.png', generatePNGIcon(512));
    console.log('✅ Generated icon-192.png and icon-512.png');
} catch (error) {
    // Canvas not available, create SVG files instead
    console.log('Canvas not available. Creating SVG files instead.');
    console.log('You can convert these to PNG using an online tool or ImageMagick.');
    
    fs.writeFileSync('icon-192.svg', createSVGIcon(192));
    fs.writeFileSync('icon-512.svg', createSVGIcon(512));
    
    console.log('✅ Generated icon-192.svg and icon-512.svg');
    console.log('💡 To convert to PNG:');
    console.log('   - Use an online SVG to PNG converter');
    console.log('   - Or use ImageMagick: convert icon-192.svg icon-192.png');
    console.log('   - Or open generate-icons.html in your browser to generate PNG files');
}

File: CapstoneSystemV4/pwa-sample/generate-icons.py
#!/usr/bin/env python3
"""
Generate PWA icons
Requires: pip install Pillow
Run: python generate-icons.py
"""

try:
    from PIL import Image, ImageDraw, ImageFont
    import os
    
    def create_icon(size):
        # Create image with gradient background
        img = Image.new('RGB', (size, size), color='#667eea')
        draw = ImageDraw.Draw(img)
        
        # Create gradient effect (simplified)
        for i in range(size):
            ratio = i / size
            r1, g1, b1 = 102, 126, 234  # #667eea
            r2, g2, b2 = 118, 75, 162   # #764ba2
            r = int(r1 + (r2 - r1) * ratio)
            g = int(g1 + (g2 - g1) * ratio)
            b = int(b1 + (b2 - b1) * ratio)
            draw.rectangle([(0, i), (size, i+1)], fill=(r, g, b))
        
        # Draw rocket emoji (as text)
        try:
            # Try to use a font that supports emoji
            font_size = int(size * 0.4)
            font = ImageFont.truetype("arial.ttf", font_size)
        except:
            try:
                font = ImageFont.truetype("/System/Library/Fonts/Apple Color Emoji.ttc", font_size)
            except:
                font = ImageFont.load_default()
        
        # Draw white circle as background for emoji
        circle_size = int(size * 0.6)
        circle_pos = ((size - circle_size) // 2, (size - circle_size) // 2)
        draw.ellipse([circle_pos[0], circle_pos[1], 
                     circle_pos[0] + circle_size, circle_pos[1] + circle_size], 
                    fill='white')
        
        # Draw rocket (simplified - just a triangle)
        triangle_points = [
            (size // 2, size // 2 - size // 4),  # Top
            (size // 2 - size // 6, size // 2 + size // 6),  # Bottom left
            (size // 2 + size // 6, size // 2 + size // 6),  # Bottom right
        ]
        draw.polygon(triangle_points, fill='#2196F3')
        
        return img
    
    # Generate icons
    icon_192 = create_icon(192)
    icon_192.save('icon-192.png', 'PNG')
    print('✅ Generated icon-192.png')
    
    icon_512 = create_icon(512)
    icon_512.save('icon-512.png', 'PNG')
    print('✅ Generated icon-512.png')
    
except ImportError:
    print("Pillow not installed. Install it with: pip install Pillow")
    print("\nAlternatively, use generate-icons.html in your browser to generate the icons.")
except Exception as e:
    print(f"Error: {e}")
    print("\nYou can use generate-icons.html in your browser instead.")

